pkg/templateservicebroker/servicebroker/provision.go

```diff
@@ -42,8 +42,14 @@ func (b *Broker) ensureSecret(u user.Info, namespace string, brokerTemplateInsta
 		Data: map[string][]byte{},
 	}
 
+	// allow empty string values for non-generated parameters.
+	// empty string values will be ignored for generated parameter/
 	for k, v := range preq.Parameters {
-		secret.Data[k] = []byte(v)
+		for _, param := range template.Parameters {
+			if param.Name == k && ((v == "" && param.Generate == "") || v != "") {
+				secret.Data[k] = []byte(v)
```

Comment:
have you confirmed that setting emptystring here ultimately results in the default value for the parameter being used during instantiation?

Reply:
@bparees I was able to verify the following
- if the default values were left alone, it's passed along as-is
- if the default values were removed/deleted, the parameters were set to an empty value 
- empty values for the generated parameters were indeed generated, where it did not before.  

However, I had difficulty getting any of the provisions to successfully complete with the cluster that @spadgett was nice enough to let me test with.

I'll change this PR as WIP, until I can fully verify a successful provision.

Reply:
> However, I had difficulty getting any of the provisions to successfully complete with the cluster that @spadgett was nice enough to let me test with.
> 
> I'll change this PR as WIP, until I can fully verify a successful provision.

It looks like react-jsonschema-form leaves out string values that are cleared by the user. The empty string isn't treated as an error for required properties in JSON schema, so they switched to undefined.

https://github.com/mozilla-services/react-jsonschema-form/pull/442

I'll have to look at how to handle this from the client.

Reply:
> if the default values were removed/deleted, the parameters were set to an empty value

Just to clarify, if the parameter is missing entirely from the request, we'd still want to use the default value. We'd only want this behavior if it's explicitly set to `""` (i.e., the user explicitly cleared the value in the web console form)

@bparees Does that make sense to you?

Reply:
yes

Reply:
@johnkim76 can you please add unit test to exercise this?

Reply:
@mfojtik will do